### PR TITLE
Updated `admission-webhook`

### PR DESCRIPTION
**Pull request by TakoB222**
### Description
---
- Rolled back source-subdir
- Added tests
- Added tox.ini

### CODE REVIEW

File: admission-webhook/rockcraft.yaml [hunk 1]
```diff
@@ -28,24 +28,23 @@ parts:
...
+      - go/1.21/stable
     source: https://github.com/kubeflow/kubeflow.git
-    source-tag: v1.7-branch 
+    source-tag: v1.9.0
+    source-depth: 1
+    source-subdir: components/admission-webhook
     build-environment:
       - CGO_ENABLED: 0
       - GOOS: linux
     override-build: |
```
Comment (kimwnasptd) [inline admission-webhook/rockcraft.yaml:41]: Can we 
1. Switch to using the `go` plugin
2. fully omit the `override-build`
3. Have the go plugin build the file?

File: admission-webhook/rockcraft.yaml [hunk 2]
```diff
@@ -28,24 +28,23 @@ parts:
...
-      cd components/admission-webhook 
-      go build -o webhook -a .
-      cp webhook $CRAFT_PART_INSTALL/webhook
-    organize:
```
Comment (kimwnasptd) [inline admission-webhook/rockcraft.yaml:43]: Why did we remove the `organize` instruction here that move the binary?

File: admission-webhook/rockcraft.yaml (hunk 1 above)
Comment (TakoB222) [reply admission-webhook/rockcraft.yaml:41]: `go` plugin is not working for me. When I added tests, find out that it's not building binary

File: admission-webhook/rockcraft.yaml (hunk 2 above)
Comment (TakoB222) [reply admission-webhook/rockcraft.yaml:43]: I can roll back it, just did to keep similar to other rocks

File: admission-webhook/rockcraft.yaml (hunk 1 above)
Comment (kimwnasptd) [reply admission-webhook/rockcraft.yaml:41]: @TakoB222 the following yaml is working for me

```yaml
name: admission-webhook
base: ubuntu@20.04
version: v1.9.0
summary: An image for Kubeflow's admission-webhook
description: |
  Admission webhook controller in general, intercepts requests to the Kubernetes API server,
  and can modify and/or validate the requests. Here the admission webhook is implemented to
  modify pods based on the available PodDefaults. When a pod creation request is received,
  the admission webhook looks up the available PodDefaults which match the pod's label.
  It then, mutates the Pod spec according to PodDefault's spec.
license: Apache-2.0

platforms:
  amd64:

services:
  base-admission-webhook:
    override: merge
    command: "/webhook"
    startup: enabled
    user: ubuntu

parts:
  admission-webhook:
    plugin: go
    build-snaps:
      - go/1.21/stable
    source: https://github.com/kubeflow/kubeflow.git
    source-tag: v1.9.0
    source-depth: 1
    source-subdir: components/admission-webhook
```

Comment (kimwnasptd) [reply admission-webhook/rockcraft.yaml]: @TakoB222 can you also include the previous sections for the ubuntu user? The above was more of a PoC to confirm the go plugin works

Comment (TakoB222) [reply admission-webhook/rockcraft.yaml]: @kimwnasptd Yes, commit with fix that you suggested already include sections for the ubuntu user

**Review by kimwnasptd — APPROVED** [review]
Thanks for the PR!

**Pull request merged**